aws/resource_aws_lb_target_group_attachment.go

```diff
@@ -140,6 +140,19 @@ func resourceAwsLbAttachmentRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error reading Target Health: %s", err)
 	}
 
+	for _, targetDesc := range resp.TargetHealthDescriptions {
+		if *targetDesc.Target.Id == d.Get("target_id").(string) {
```

Comment:
To prevent potential panics we should perform `nil` checking and use the AWS Go SDK provided conversion function instead of directly dereferencing, e.g.

```suggestion
		if targetDesc == nil && targetDesc.Target == nil {
			continue
		}

		if aws.StringValue(targetDesc.Target.Id) == d.Get("target_id").(string) {
```